dali/operators/decoder/video/video_decoder_cpu.cc

```diff
@@ -12,66 +12,87 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "dali/operators/decoder/video/video_decoder_cpu.h"
+#include "dali/operators/decoder/video/video_decoder_base.h"
+#include "dali/operators/reader/loader/video/frames_decoder.h"
 
 namespace dali {
 
-bool VideoDecoderCpu::SetupImpl(std::vector<OutputDesc> &output_desc,
-                                const Workspace &ws) {
-  ValidateInput(ws);
-  const auto &input = ws.Input<CPUBackend>(0);
-  int batch_size = input.num_samples();
-  frames_decoders_.resize(batch_size);
-  auto &thread_pool = ws.GetThreadPool();
-  for (int i = 0; i < batch_size; ++i) {
-    thread_pool.AddWork([this, i, &input](int tid) {
-      auto sample = input[i];
-      auto data = reinterpret_cast<const char *>(sample.data<uint8_t>());
-      size_t size = sample.shape().num_elements();
-      auto source_info = input.GetMeta(i).GetSourceInfo();
-      frames_decoders_[i] = std::make_unique<FramesDecoder>(data, size, false, true, -1,
-                                                            source_info);
-    });
-  }
-  thread_pool.RunAll();
-  for (auto &dec : frames_decoders_) {
-    DALI_ENFORCE(dec->IsValid(), make_string("Failed to create video decoder for \"",
-                                 dec->Filename(), "\""));
-  }
-  output_desc.resize(1);
-  output_desc[0].shape = ReadOutputShape();
-  output_desc[0].type = DALI_UINT8;
-  return true;
-}
-
-void VideoDecoderCpu::RunImpl(Workspace &ws) {
-  auto &output = ws.Output<CPUBackend>(0);
-  const auto &input = ws.Input<CPUBackend>(0);
-  int batch_size = input.num_samples();
-  auto &thread_pool = ws.GetThreadPool();
-  for (int s = 0; s < batch_size; ++s) {
-    thread_pool.AddWork([this, s, &output](int tid) {
-      DecodeSample(output[s], s);
-    }, volume(input[s].shape()));
-  }
-  thread_pool.RunAll();
-}
-
 DALI_SCHEMA(experimental__decoders__Video)
     .DocStr(
-        R"code(Decodes a video file from a memory buffer (e.g. provided by external source).
+        R"code(Decodes video files from memory buffers into sequences of frames.
+
+The operator accepts video files in common container formats (e.g. MP4, AVI). For CPU backend,
+FFmpeg is used for decoding. For Mixed backend, NVIDIA's Video Codec SDK (NVDEC) is used.
+
+Each output sample is a sequence of frames with shape (F, H, W, C) where:
+- F is the number of frames in the sequence (can vary between samples)
+- H is the frame height in pixels
+- W is the frame width in pixels 
+- C is the number of color channels
```

Comment:
This part doesn't render properly
![image](https://github.com/user-attachments/assets/e6352e3c-6b2c-4a55-b8bb-bd2f8b724e5e)